src/dstack/_internal/proxy/gateway/model_routers/base.py

```
@@ -0,0 +1,91 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import List, Literal, Optional
+
+from pydantic import BaseModel
+
+from dstack._internal.core.models.routers import AnyRouterConfig
+
+
+class RouterContext(BaseModel):
+    """Context for router initialization and configuration."""
+
+    class Config:
+        frozen = True
+
+    host: str = "127.0.0.1"
+    port: int = 3000
+    log_dir: Path = Path("./router_logs")
```

(nit) These defaults here are no longer relevant after switching to per-service routers and dynamic port allocation.

I can suggest to remove the defaults to avoid misleading the devs. And then explicit `RouterContext` will also become required in [`Router.__init__`](https://github.com/dstackai/dstack/blob/71cacf2a53d6a818a902dc2e3a1bfbf5f3b284da/src/dstack/_internal/proxy/gateway/model_routers/base.py#L32)

---

Yes router_port is only set when a router exists and only sglang router exists currently.

---

Resolved